nextcloudappstore/core/models.py

```diff
@@ -88,34 +89,52 @@ def can_update(self, user: User) -> bool:
     def can_delete(self, user: User) -> bool:
         return self.owner == user
 
-    def releases_by_platform_v(self):
+    def releases_by_platform_v(self, nightlies=False):
         """Looks up all compatible releases for each platform version.
+
+        :param nightlies: Return only nightly releases if true, and only non-
+                          nightlies otherwise.
         :return dict with all compatible releases for each platform version.
         """
+
         releases = {}
         for p_version in settings.PLATFORM_VERSIONS:
-            releases[p_version] = self.compatible_releases(p_version)
+            releases[p_version] = \
+                    self.compatible_releases(p_version, nightlies=nightlies)
         return releases
 
     def latest_releases_by_platform_v(self):
         """Looks up the latest release for each platform version.
         Ignores nightly releases.
         :return dict with the latest release for each platform version.
         """
+
         all_latest = {}
         for p_version in settings.PLATFORM_VERSIONS:
             compatible = self.compatible_releases(p_version)
             all_latest[p_version] = self._latest_non_nightly(compatible)
         return all_latest
 
-    def compatible_releases(self, platform_version, inclusive=True):
+    def compatible_releases(self, platform_version, inclusive=True,
+                            nightlies=False):
+        """Returns all releases of this app that are compatible with the given
+        platform version.
+
+        :param inclusive: Use inclusive version check (see
+                          AppRelease.is_compatible()).
+        :param nightlies: Return only nightly releases if True, only non-
+                          nightlies otherwise.
+        :return a sorted list of all compatible releases.
+        """
+
         all_releases = self.releases.all()
         return sorted(list(filter(
```

Comment:
Then lets remove list()

list() forces the iterator to be evaluated which is not only slow but also most of the time unneeded :)